### ISSUE TITLE
survreg NA error for data sets with open intervals

### ISSUE THREAD

**Issue by tdhock** [therneau/survival#8]
Hey @therneau how's it going? I am a big fan of your survreg function, which I would like to discuss in a tutorial about learning penalty functions for optimal change-point detection. https://github.com/tdhock/change-tutorial/blob/master/Submission.md

In the real data I have only open intervals (-Inf, a) or (b, Inf) in my Surv object. When I try to perform a regression on one variable, I get an error:

```r
> fit <- survreg(Surv(min.L, max.L, type="interval2") ~ feature, train.df, dist="gaussian")
Error in coxph.wtest(t(x) %*% (wt * x), c((wt * eta + weights * deriv$dg) %*%  : 
  NA/NaN/Inf in foreign function call (arg 3)
```
I think this error is rather strange, because it says something about "coxph" but I don't want to fit a Cox model! I wanted to fit a parametric AFT model using the Gaussian loss function. Am I mis-understanding something about how to use `survreg`?

I managed to create a minimal reproducible example (MRE) which you can see here https://gist.github.com/tdhock/f0d2d5991232b932cbc2426d226b4ecf#file-toy-intervals-r along with the R output I got, and the package versions that I used. The MRE contains only 5 data points, 4 of them are open intervals, and 1 of them is a closed interval. Is there some problem when there are no un-censored observations at all?

Thanks in advance for checking this out.

**Comment by therneau** [comment]
You are correct, it is a bug.  Fixed in version 2.41-0, which I am currently testing out.  Hope to push it to github in the next couple days.  A grant deadline for last week put me behind on everything else.

**Comment by tdhock** [comment]
thanks very much @therneau
